engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java

```diff
@@ -458,6 +459,16 @@ protected Answer migrateVolumeToPool(DataObject srcData, DataObject destData) {
         StoragePool srcPool = (StoragePool)dataStoreMgr.getDataStore(srcData.getDataStore().getId(), DataStoreRole.Primary);
         StoragePool destPool = (StoragePool)dataStoreMgr.getDataStore(destData.getDataStore().getId(), DataStoreRole.Primary);
         MigrateVolumeCommand command = new MigrateVolumeCommand(volume.getId(), volume.getPath(), destPool, volume.getAttachedVmName(), volume.getVolumeType(), waitInterval, volume.getChainInfo());
+
+        if (volume.getpayload() instanceof DiskOfferingVO) {
+            DiskOfferingVO offering = (DiskOfferingVO) volume.getpayload();
+            Long newIops = null;
+            if (offering.getIopsReadRate() != null && offering.getIopsWriteRate() != null) {
+                newIops = offering.getIopsReadRate() + offering.getIopsWriteRate();
+            }
+            command.setNewIops(newIops);
```

Comment:
only IOPS are considered, should consider other attributes in the new disk offering?

Reply:
Not that I am aware at the moment, this PR focus in the IOPS limitation, if needed we can create another PR in the future to address other attributes.